kymatio/frontend/tensorflow_frontend.py

```diff
@@ -0,0 +1,20 @@
+import tensorflow as tf
+from packaging import version
+
+assert version.parse(tf.__version__)>=version.parse("2.0.0a0"), 'Current TensorFlow version is '+str(tf.__version__)+\
```

Comment:
Do we want to be supporting kymatio for only the most recent version of tf? Why not issue a warning instead? 

I've also heard that people prefer the older versions of TF. Might impact the usage of this software package.

Reply:
yes, we do, trust me 🤓

Reply:
1/ The APIs are changing quite quickly, it's pretty hard to be compatible with both(I didn't want to spend time on that) I think we should only support the latest vrsion of torch or tf, it's easier.
2/ We can't add this as a `requirements.txt` element because it is optional.
3/ a really motivated use could add its own backend for a given tf version. We might be willing to do that later but it is really too much work for now (I spend a significant time just to make sure the same version of tf was working for python 3.5, 3.6 and 3.7)

Reply:
I remember Pytorch 1.2 and 1.3 worked for the changes we made (I'll check how earlier versions of Pytorch work with our update). 

My thinking is, if it isn't broken, let the users use it. If we get to a point where we need to use some new Pytorch/Tensorflow functionality, then we have a backend that works for the past versions, and we can simply copy the old backend and update. 

I suppose that comes with the disadvantage that we may have to make sure that our software works for all versions of Pytorch/Tensorflow that we intend to support. Thoughts @janden ?

Reply:
Ok, for (1), I agree. It seems that TF1 and TF2 are quite different beasts. Now with torch, it's not clear to me that we can't support earlier versions (at least down to 1.0) without much effort. If it requires a bunch of extra work, sure, let's restrict it to 1.3, but if not, I don't see why we can't let users use an earlier version. We have to bear in mind that users have their constraints too, let's not burden them with more if it's not needed.

For (2), we used to have a bunch of different `requirements.txt` for different purposes (like `requirements_skcuda.txt`). Why not stick to that?

For (3) sure, why not?

Reply:
I think deep learning pipelines change too fastly. The problem is not to use new functionality, yet the syntax is simply different. This was the case with `tensorflow`. Also for debugging purposes, I think it's better to constraint the version we're using. PyTorch got recently updated, why not setting this as a breaking point?

Reply:
for (1): right - I'd prefer to set the flag at the latest torch version but why not. I have no idea which version breaks or not, but if one decided to include more version, then each of them should be included in the unit tests.
for (2): it is not required to have cuda, so i don't understand why this should be a requirement

Reply:
(1) I don't mind setting up the testing for different versions. Better to be safe than sorry. Also, from what I've seen, the PyTorch people do seem quite careful about maintaining backwards compatbility (at least more so than TF), so I don't think it's too big of a risk.

(2) I'm not saying it should be a requirement. I'm saying that we should stick the packages in a file (so that we can install them with `pip install -r`), that way if we add packages for that backend later on, we're just updating that file, not the docs (where it may occur in several places). It also makes updating the tests easier, since they're just using that file.

Reply:
for (2), see with @eickenberg